```
 Did you mean '/*'?
/home/runner/work/fineract/fineract/fineract-provider/src/main/java/org/apache/fineract/portfolio/note/service/NoteWritePlatformServiceJpaRepositoryImpl.java:170: error: [UnusedMethod] Method 'updateSavingAccountNote' is never used.
    private CommandProcessingResult updateSavingAccountNote(final JsonCommand command) {
                                    ^
    (see https://errorprone.info/bugpattern/UnusedMethod)
  Did you mean to remove this line?
/home/runner/work/fineract/fineract/fineract-provider/src/main/java/org/apache/fineract/portfolio/note/service/NoteWritePlatformServiceJpaRepositoryImpl.java:128: error: [UnusedMethod] Method 'updateLoanNote' is never used.
    private CommandProcessingResult updateLoanNote(final JsonCommand command) {
                                    ^
    (see https://errorprone.info/bugpattern/UnusedMethod)
  Did you mean to remove this line?
/home/runner/work/fineract/fineract/fineract-provider/src/main/java/org/apache/fineract/portfolio/note/service/NoteWritePlatformServiceJpaRepositoryImpl.java:148: error: [UnusedMethod] Method 'updateLoanTransactionNote' is never used.
    private CommandProcessingResult updateLoanTransactionNote(final JsonCommand command) {
```
    
    
    Please rebase, and review the failing checks.

The code not compiling either due to errors (see above).

To validate please run locally:
`./gradlew spotlessApply spotbugsMain spotbugsTest checkstyleMain checkstyleTest`
`./gradlew --no-daemon build -x test -x cucumber -x doc`

--- a/fineract-core/src/main/java/org/apache/fineract/portfolio/calendar/domain/Calendar.java
+++ b/fineract-core/src/main/java/org/apache/fineract/portfolio/calendar/domain/Calendar.java
@@ -518,7 +518,7 @@ public class Calendar extends AbstractAuditableWithUTCDateTimeCustom<Long> {
         final StringBuilder recurrenceBuilder = new StringBuilder(200);
 
         recurrenceBuilder.append("FREQ=");
-        recurrenceBuilder.append(frequencyType.toString().toUpperCase());
+        recurrenceBuilder.append(frequencyType.toString().toUpperCase(java.util.Locale.ROOT));
         if (interval > 1) {
             recurrenceBuilder.append(";INTERVAL=");
             recurrenceBuilder.append(interval);
@@ -528,7 +528,7 @@ public class Calendar extends AbstractAuditableWithUTCDateTimeCustom<Long> {
                 final CalendarWeekDaysType weekDays = CalendarWeekDaysType.fromInt(repeatsOnDay);
                 if (!weekDays.isInvalid()) {
                     recurrenceBuilder.append(";BYDAY=");
-                    recurrenceBuilder.append(weekDays.toString().toUpperCase());
+                    recurrenceBuilder.append(weekDays.toString().toUpperCase(java.util.Locale.ROOT));
                 }
             }
         }
@@ -548,7 +548,7 @@ public class Calendar extends AbstractAuditableWithUTCDateTimeCustom<Long> {
                 final CalendarWeekDaysType weekday = CalendarWeekDaysType.fromInt(repeatsOnDay);
                 if (!weekday.isInvalid()) {
                     recurrenceBuilder.append(";BYDAY=");
-                    recurrenceBuilder.append(weekday.toString().toUpperCase());
+                    recurrenceBuilder.append(weekday.toString().toUpperCase(java.util.Locale.ROOT));
                 }
             }
         }

--- a/fineract-core/src/main/java/org/apache/fineract/portfolio/calendar/service/CalendarUtils.java
+++ b/fineract-core/src/main/java/org/apache/fineract/portfolio/calendar/service/CalendarUtils.java
@@ -299,10 +299,11 @@ public final class CalendarUtils {
                 NthDayNameEnum nthDayName = NthDayNameEnum.from(nthDayType.toString());
                 DayNameEnum weekdayType = DayNameEnum.from(weekDay.getDay().name());
                 if (recur.getInterval() == 1 || recur.getInterval() == -1) {
-                    humanReadable = "Monthly on " + nthDayName.getCode().toLowerCase() + " " + weekdayType.getCode().toLowerCase();
+                    humanReadable = "Monthly on " + nthDayName.getCode().toLowerCase(java.util.Locale.ROOT) + " "
+                            + weekdayType.getCode().toLowerCase(java.util.Locale.ROOT);
                 } else {
-                    humanReadable = "Every " + recur.getInterval() + " months on " + nthDayName.getCode().toLowerCase() + " "
-                            + weekdayType.getCode().toLowerCase();
+                    humanReadable = "Every " + recur.getInterval() + " months on " + nthDayName.getCode().toLowerCase(java.util.Locale.ROOT)
+                            + " " + weekdayType.getCode().toLowerCase(java.util.Locale.ROOT);
                 }
             } else if (monthDay != null) {
                 if (monthDay == -1) {
@@ -756,7 +757,8 @@ public final class CalendarUtils {
             if (nthDayType == NthDayType.ONE || nthDayType == NthDayType.TWO || nthDayType == NthDayType.THREE
                     || nthDayType == NthDayType.FOUR) {
                 baseDataValidator.reset().parameter(repeatsOnDayParamName).value(repeatsOnDay).cantBeBlankWhenParameterProvidedIs(
-                        repeatsOnNthDayOfMonthParamName, NthDayNameEnum.from(nthDayType.toString()).getCode().toLowerCase());
+                        repeatsOnNthDayOfMonthParamName,
+                        NthDayNameEnum.from(nthDayType.toString()).getCode().toLowerCase(java.util.Locale.ROOT));
             }
         }
     }

--- a/fineract-core/src/main/java/org/apache/fineract/useradministration/domain/AppUser.java
+++ b/fineract-core/src/main/java/org/apache/fineract/useradministration/domain/AppUser.java
@@ -552,8 +552,9 @@ public class AppUser extends AbstractPersistableCustom<Long> implements Platform
     }
 
     private void validateHasPermission(final String prefix, final String resourceType) {
-        final String authorizationMessage = "User has no authority to " + prefix + " " + resourceType.toLowerCase() + "s";
-        final String matchPermission = prefix + "_" + resourceType.toUpperCase();
+        final String authorizationMessage = "User has no authority to " + prefix + " " + resourceType.toLowerCase(java.util.Locale.ROOT)
+                + "s";
+        final String matchPermission = prefix + "_" + resourceType.toUpperCase(java.util.Locale.ROOT);
 
         if (!hasNotPermissionForAnyOf("ALL_FUNCTIONS", "ALL_FUNCTIONS_READ", matchPermission)) {
             return;
@@ -636,7 +637,7 @@ public class AppUser extends AbstractPersistableCustom<Long> implements Platform
     }
 
     public void validateHasCheckerPermissionTo(final String function) {
-        final String checkerPermissionName = function.toUpperCase() + "_CHECKER";
+        final String checkerPermissionName = function.toUpperCase(java.util.Locale.ROOT) + "_CHECKER";
         if (hasNotPermissionTo("CHECKER_SUPER_USER") && hasNotPermissionTo(checkerPermissionName)) {
             final String authorizationMessage = "User has no authority to be a checker for: " + function;
             throw new NoAuthorizationException(authorizationMessage);

--- a/fineract-provider/src/main/java/org/apache/fineract/infrastructure/campaigns/sms/domain/SmsCampaign.java
+++ b/fineract-provider/src/main/java/org/apache/fineract/infrastructure/campaigns/sms/domain/SmsCampaign.java
@@ -523,7 +523,7 @@ public class SmsCampaign extends AbstractPersistableCustom<Long> {
         final StringBuilder recurrenceBuilder = new StringBuilder(200);
 
         recurrenceBuilder.append("FREQ=");
-        recurrenceBuilder.append(frequencyType.toString().toUpperCase());
+        recurrenceBuilder.append(frequencyType.toString().toUpperCase(java.util.Locale.ROOT));
         if (interval > 1) {
             recurrenceBuilder.append(";INTERVAL=");
             recurrenceBuilder.append(interval);
@@ -533,7 +533,7 @@ public class SmsCampaign extends AbstractPersistableCustom<Long> {
                 final CalendarWeekDaysType weekDays = CalendarWeekDaysType.fromInt(repeatsOnDay);
                 if (!weekDays.isInvalid()) {
                     recurrenceBuilder.append(";BYDAY=");
-                    recurrenceBuilder.append(weekDays.toString().toUpperCase());
+                    recurrenceBuilder.append(weekDays.toString().toUpperCase(java.util.Locale.ROOT));
                 }
             }
         }

--- a/fineract-validation/src/main/java/org/apache/fineract/validation/constraints/EnumValueValidator.java
+++ b/fineract-validation/src/main/java/org/apache/fineract/validation/constraints/EnumValueValidator.java
@@ -30,12 +30,12 @@ public class EnumValueValidator implements ConstraintValidator<EnumValue, String
 
     @Override
     public void initialize(EnumValue annotation) {
-        acceptedValues = Arrays.stream(annotation.enumClass().getEnumConstants()).map(e -> e.name().toLowerCase())
+        acceptedValues = Arrays.stream(annotation.enumClass().getEnumConstants()).map(e -> e.name().toLowerCase(java.util.Locale.ROOT))
                 .collect(Collectors.toSet());
     }
 
     @Override
     public boolean isValid(String value, ConstraintValidatorContext context) {
-        return value != null && acceptedValues.contains(value.toLowerCase());
+        return value != null && acceptedValues.contains(value.toLowerCase(java.util.Locale.ROOT));
     }
 }

